media-proxy/src/proxy_context.cc

```diff
@@ -787,23 +787,59 @@ int ProxyContext::TxStart(const TxControlRequest* request)
         return -1;
     }
 
-    st_ctx = new (mtl_session_context_t);
+    st_ctx = new (dp_session_context_t);
     st_ctx->id = incrementMSessionCount();
     st_ctx->type = TX;
     st_ctx->tx_session = tx_ctx;
-    mStCtx.push_back(st_ctx);
+    mDpCtx.push_back(st_ctx);
 
     /* TODO: to be removed later. */
     mTxCtx.push_back(tx_ctx);
 
     return (st_ctx->id);
 }
 
-int ProxyContext::RxStart(const mcm_conn_param* request)
+int ProxyContext::RxStart_rdma(const mcm_conn_param* request) {
+    rdma_s_ops_t opts = { 0 };
+    dp_session_context_t* dp_ctx = NULL;
+    memif_ops_t memif_ops = { 0 };
+    rx_rdma_session_context_t* rx_ctx = NULL;
+    int ret;
+    if (mDevHandle_rdma == NULL) {
+        ret = rdma_init(&mDevHandle_rdma);
+        if (ret) {
+            INFO("%s, Fail to initialize libfabric.\n", __func__);
+            return -1;
```

Comment:
Use some standard error code to return an error.
```suggestion
            return -EINVAL;
```

Reply:
We should define better error codes or use those from errno.h for the whole mcm proxy. "-1" is used everywhere.

Reply:
Yeah, I suggest using standard error codes in new features instead of keeping returning -1. We'll schedule a bigger or even global refactoring in the future development cycles.

Reply:
Ok, I will change it to EINVAL, but honestly, it does not give more context. We can talk about improving error handling in the future.